content/docs/for-buildpack-authors/how-to/write-buildpacks/specify-env.md

```diff
@@ -3,15 +3,108 @@ title="Specify the environment"
+
 <!--more-->
 
-This page is a stub! The CNB project is applying to [Google Season of Docs](https://developers.google.com/season-of-docs/docs/timeline) to receive support for improving our documentation. Please check back soon.
+### POSIX Path Variables
+
+The following layer path environment variables MUST be set by the `lifecycle` during the `build` and `launch` phases in order to make buildpack dependencies accessible.
+
+During the `build` phase, each variable designated for build MUST contain absolute paths of all previous buildpacks’ `<layers>/<layer>/` directories that are designated for build.
+
+When the exported OCI image is launched, each variable designated for launch MUST contain absolute paths of all buildpacks’ `<layers>/<layer>/` directories that are designated for launch.
+
+In either case,
+
+- The lifecycle MUST order all `<layer>` paths to reflect the reversed order of the buildpack group.
+- The lifecycle MUST order all `<layer>` paths provided by a given buildpack alphabetically ascending.
+- The lifecycle MUST separate each path with the OS path list separator (e.g. `:` on Linux, `;` on Windows).
+
+| Env Variable                               | Layer Path   | Contents         | Build | Launch |
+|--------------------------------------------|--------------|------------------|-------|--------|
+| `PATH`                                     | `/bin`       | binaries         | [x]   | [x]    |
+| `LD_LIBRARY_PATH`                          | `/lib`       | shared libraries | [x]   | [x]    |
+| `LIBRARY_PATH`                             | `/lib`       | static libraries | [x]   |        |
+| `CPATH`                                    | `/include`   | header files     | [x]   |        |
+| `PKG_CONFIG_PATH`                          | `/pkgconfig` | pc files         | [x]   |        |
+
+The platform SHOULD NOT assume any other base-image-provided environment variables are inherited by the buildpack.
+
+### Build Environment Variables
+
+During the `build` phase, buildpacks MAY write environment variable files to `<layers>/<layer>/env/`, `<layers>/<layer>/env.build/`, and `<layers>/<layer>/env.launch/` directories.
+
+For each `<layers>/<layer>/` designated as a build layer, for each file written to `<layers>/<layer>/env/` or `<layers>/<layer>/env.build/` by `/bin/build`, the `lifecycle` MUST modify an environment variable in subsequent executions of `/bin/build` according to the [environment variable modification rules](https://github.com/buildpacks/spec/blob/main/buildpack.md#environment-variable-modification-rules).
+
+For each file written to `<layers>/<layer>/env/` or `<layers>/<layer>/env.launch/` by `/bin/build`, the `lifecycle` MUST modify an environment variable during the launch phase according to the modification rules above.
+
+### Process Environment Variables
+
+TBA
+
+### Runtime Environment Variables
+
+TBA
+
+### Environment Variable Modification Rules
```

Comment:
I think some specific examples would help here. We could use this tree:

```
layers/
├── some-buildpack-id
│   ├── some-layer
│   │   └── env
│   │       ├── SOME_VAR # contents foo
│   │       └── SOME_VAR.append # contents :
│   └── some-layer.toml
└── some-other-buildpack-id
    ├── some-layer
    │   └── env
    │       ├── SOME_VAR # contents bar
    │       └── SOME_VAR.append # contents :
    └── some-layer.toml
```

And walk through what would be the final value of `SOME_VAR` for each of `append`, `default`, `override`, and `prepend`. Hopefully the comments above provide enough context to construct these examples, but LMK if you get stuck here.

Reply:
@natalieparellano what would be the final value of SOME_VAR? not sure I understand this

Reply:
It would be `foo:bar`

Here is the relevant part of the spec:

> Append
The value of the environment variable MUST be a concatenation of the file contents and the contents of other files representing that environment variable. Within that environment variable value,
> * Earlier buildpacks' environment variable file contents MUST precede later buildpacks' environment variable file contents.

> Delimiter
The file contents MUST be used to delimit any concatenation within the same layer involving that environment variable.

Reply:
added

Reply:
done